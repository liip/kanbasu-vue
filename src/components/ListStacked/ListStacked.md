```vue
<list-stacked size="large" divided>
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
</list-stacked>
```