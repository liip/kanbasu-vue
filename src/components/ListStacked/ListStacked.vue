<template>
  <ul :class="classNames">
    <slot></slot>
  </ul>
</template>

<script>
export default {
  name: 'list-stacked',

  props: {
    /**
     * List items gutter size:
     * `tight, tiny, small, large, huge`
     */
    size: {
      type: String,
    },
    /**
     * Add separators between items
     */
    divided: {
      type: Boolean,
    },
  },

  computed: {
    classNames() {
      return [
        'list-stacked',
        { [`list-stacked--${this.size}`]: !!this.size },
        { 'list-stacked--divided': this.divided },
      ];
    },
  },
};
</script>
