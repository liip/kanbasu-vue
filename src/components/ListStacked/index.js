import ListStacked from './ListStacked.vue';

export default ListStacked;
