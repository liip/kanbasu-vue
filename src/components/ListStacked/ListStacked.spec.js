import { shallowMount } from '@vue/test-utils';
import ListStacked from './ListStacked.vue';

describe('ListStacked.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(ListStacked);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with size', () => {
    const wrapper = shallowMount(ListStacked, {
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with dividers', () => {
    const wrapper = shallowMount(ListStacked, {
      propsData: {
        divided: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
