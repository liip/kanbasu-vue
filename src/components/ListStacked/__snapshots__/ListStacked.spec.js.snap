// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListStacked.vue renders 1`] = `<ul class="list-stacked"></ul>`;

exports[`ListStacked.vue renders with dividers 1`] = `<ul class="list-stacked list-stacked--divided"></ul>`;

exports[`ListStacked.vue renders with size 1`] = `<ul class="list-stacked list-stacked--small"></ul>`;
